common/src/Model/Brush.cpp

```diff
@@ -56,17 +56,15 @@ namespace TrenchBroom {
             }
 
             void faceWasCreated(BrushFaceGeometry* face) override {
-                face->setPayload(m_addedFace);
                 m_addedFace->setGeometry(face);
             }
 
             void faceWillBeDeleted(BrushFaceGeometry* face) override {
-                BrushFace* brushFace = face->payload();
+                auto* brushFace = face->payload();
                 if (brushFace != nullptr) {
                     ensure(!brushFace->selected(), "brush face is selected");
-
+                    brushFace->setGeometry(nullptr);
```

Comment:
The `setGeometry(nullptr);` can be moved to `~BrushFace`, I think?

Reply:
I thought about that, but since `BrushFace::setGeometry` will call `Polyhedron::Face::setPayload`, I am afraid of getting issues with the destruction order. I think it's too risky.

Reply:
Ah, that's fair.

As a later change, `~Polyhedron::Face` could clear the pointer in BrushFace (e.g. it calls a function `m_payload->polyhedronFaceIsDestructing(Polyhedron::Face* face)` on the payload), then clearing these weak references can be fully handled within both destructors.